src/main/java/frc/robot/Robot.java

```diff
@@ -25,18 +35,31 @@
  */
 public class Robot {
 
+	private final SwerveDriveSimulation swerveDriveSimulation;
+
 	public static final RobotType ROBOT_TYPE = RobotType.determineRobotType();
 
 	private final Swerve swerve;
 	private final PoseEstimator poseEstimator;
 	private final Superstructure superStructure;
 
 	public Robot() {
+		GyroSimulation gyroSimulation = null;
+		if (ROBOT_TYPE.isSimulation()) {
+			gyroSimulation = SimulationGyroConstants.generateGyroSimulation();
+			Supplier<SwerveModuleSimulation> simulationModule = SimulationModuleGenerator.generate();
+			this.swerveDriveSimulation = SimulationSwerveGenerator
+				.generate(simulationModule, gyroSimulation, PoseEstimatorConstants.DEFAULT_POSE);
+			SimulatedArena.getInstance().addDriveTrainSimulation(swerveDriveSimulation);
+		} else {
+			swerveDriveSimulation = null;
```

Comment:
Use optional instead, and pass it as an argument to the factory.

Reply:
Now I do it but without optional

Reply:
There is no need for this here, the value can simply stay null as it is from the start. You simply don't need the else section

Reply:
Because of the final the else is needed